realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1319,8 +1319,8 @@ private void checkMaxDepth(int maxDepth) {
         if (realmObject == null) {
             throw new IllegalArgumentException("Null objects cannot be copied from Realm.");
         }
-        if (!RealmObject.isValid(realmObject)) {
-            throw new IllegalArgumentException("RealmObject is not valid, so it cannot be copied.");
+        if (!(RealmObject.isManaged(realmObject) && RealmObject.isValid(realmObject))) {
```

Comment:
`if (!RealmObject.isManaged(realmObject) || !RealmObject.isValid(realmObject))` is easier for me to understand. but it is fine to leave as it is.